### PR TITLE
[Refactor]use async to get be resource

### PR DESCRIPTION
**Pull request by wangbo**
## Proposed changes
Use async to get be resource to simplify rpc code and implementing async rpc

### CODE REVIEW

**Comment by wangbo** [comment]
run buildall

File: be/src/service/internal_service.cpp [hunk 1]
```diff
@@ -2105,4 +2105,22 @@ void PInternalService::get_wal_queue_size(google::protobuf::RpcController* contr
     }
 }
 
+void PInternalService::get_be_resource(google::protobuf::RpcController* controller,
+                                       const PGetBeResourceRequest* request,
+                                       PGetBeResourceResponse* response,
+                                       google::protobuf::Closure* done) {
+    brpc::ClosureGuard closure_guard(done);
+    //NOTE: currently not use a thread pool here because get mem_limit is a very lightweight operation
```
Comment (yiguolei) [inline be/src/service/internal_service.cpp:2113]: 把这个函数放到light pool 里执行，防止出现brpc 卡住的问题

**Comment by wangbo** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Comment by wangbo** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**